x-pack/legacy/plugins/security/server/routes/api/v1/authenticate.js

```diff
@@ -103,25 +134,48 @@ export function initAuthenticateApi({ authc: { login, logout }, config }, server
     },
     async handler(request, h) {
       try {
+        const query = request.query || {};
+        const payload = request.payload || {};
```

Comment:
I don't know to be honest, I was even planning to eventually get rid of query validation as well since we (Kibana) neither parse nor try to interpret them and just blindly pass to Elasticsearch anyway (and we allow unknown parameters as well). If it happens so that these args are malformed, imo, it's better to allow ES to parse them and log more detailed error in their log than having this validation in Kibana.

What do you think?

Reply:
> What do you think?

We likely get little benefit from doing it ourselves because we end up passing it directly to Elasticsearch. It could potentially help us if we received the wrong "data-type" for a specific parameter. So if we received a `payload.iss` which was an Object as opposed to a string, we could potentially have more informative error messages instead of something breaking in a somewhat obscure manner downstream.

Reply:
I'm fine with it either way, I defer to your opinion :)

Reply:
I've just tried to add a payload validation and was hit by [this limitation](https://github.com/hapijs/hapi/issues/2659) :cry: That means we can't have `payload` validation if route defines multiple methods and some of them don't support payload. The only alternative is to split this route into two very similar ones. Since you're fine with the current implementation I'll keep it simple and leave it as is for now. Let's get back to it when we migrate this route to NP (I leave a code comment).

I'll add a validation for `authenticationResponseURI` query parameter as you suggested below though.

Reply:
Sounds good!!